### PR TITLE
Setup vdkLib in project and can call udContext_Connect from Login screen

### PR DESCRIPTION
**Pull request by kressnick25**
The libraries functions that can be called are in vaultSDK/lib. Currently I've further abstracted the function call in the UdContext class, so it is called with `UdContext.connect([params])` to be able to pass regular strings as params.

This also allows us to abstract the call later, which I suspect we will have to, as running the C code currently blocks the main thread, causing the UI to freeze while the network request is completed. I tried to use the Isolate/compute async functions to solve this but you can't pass an ffi Pointer to an Isolate so have to look at it another time. 

Currently context() always responds with `ve_SecurityFailure`, this will be fixed when we can call `vdkConfig_IgnoreCertificateVerification`.

Note the vdk libs aren't included, I have appended `vautltSDK/README` to include which folders they should live in.

### CODE REVIEW

File: vaultSDK/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# vaultSDK
+
+Dart bindings for the vaultSDK
+
+## Getting Started
+
+Place all the header files from `vdk/include` in `ios/Classes/`
+
+Place the `android_arm64` and `win_x64` folders from `vdk/lib/` into a new folder `android/src/main/jniLibs`, and rename them to `arm64-v8a` and `x86_64` respectively.
```
Comment (samsinsane) [inline vaultSDK/README.md]: This should be `android_arm64` and `android_x64`.

File: vaultSDK/lib/udConfig.dart [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    int boolean = 0;
+    if (ignore) boolean = 1;
+    var err = udConfig_IgnoreCertificateVerification(boolean);
+
+    return vdkError.values[err];
+  }
+}
+
+typedef udConfig_IgnoreCertificateVerification_native = Int32 Function(
+    Int8 ignore);
```
Comment (samsinsane) [inline vaultSDK/lib/udConfig.dart:30]: When you start using udSDK, this should become `Int32 ignore`.

File: vaultSDK/lib/udContext.dart [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    return version;
+  }
+
+  // Call these functions in flutter widgets
+  static vdkError connect(Pointer<IntPtr> udContext, email, password,
+      [url = 'https://udstream.euclideon.com', appName = 'Pointfluent']) {
+    final uUrl = Utf8.toUtf8(url);
+    final uAppName = Utf8.toUtf8(appName);
+    final uEmail = Utf8.toUtf8(email);
+    final uPassword = Utf8.toUtf8(password);
```
Comment (samsinsane) [inline vaultSDK/lib/udContext.dart:24]: All four of these will leak memory unless you explicitly free that memory. The FFI package provides the `free` function for this.

**Review by samsinsane — CHANGES_REQUESTED** [review]
I would recommend once you've moved over to using udSDK, that you rename all instances of VDK and VaultSDK for consistency and to avoid confusion.

File: vaultSDK/README.md (hunk 1 above)
Comment (kressnick25) [reply vaultSDK/README.md]: Should new folder names remain the same? so `android_arm64 -> arm64-v8a` and `android_x64 -> x86_64`

File: vaultSDK/lib/udConfig.dart (hunk 2 above)
Comment (kressnick25) [reply vaultSDK/lib/udConfig.dart:30]: I've added a reminder here 468c18a8df9739835a33e20200f146ab2d1ff865

File: vaultSDK/lib/udContext.dart (hunk 3 above)
Comment (kressnick25) [reply vaultSDK/lib/udContext.dart:24]: Fixed in e5c0ab9

File: vaultSDK/README.md (hunk 1 above)
Comment (samsinsane) [reply vaultSDK/README.md]: Yes, those are the correct ABI names.

**Review by samsinsane — APPROVED** [review]

**Pull request merged**